### ISSUE TITLE
CompositeUrlRule within CompositeUrlRule

### ISSUE THREAD

**Issue by mfrancis107** [yiisoft/yii2#7296]
Getting unknown property: yii\web\GroupUrlRule::name

Composite rules do not have a name attribute. Therefore the parse request method throws an unknown property exception when you have a  

```
public function parseRequest($manager, $request)
    {
        foreach ($this->rules as $rule) {
            /* @var $rule \yii\web\UrlRule */
            if (($result = $rule->parseRequest($manager, $request)) !== false) {
                Yii::trace("Request parsed with URL rule: {$rule->name}", __METHOD__);

                return $result;
            }
        }

        return false;
    }
```

Recommend to change to. Either add a name attribute or do a check in the parseRequest method.

```
public function parseRequest($manager, $request)
    {
        foreach ($this->rules as $rule) {
            /* @var $rule \yii\web\UrlRule */
            if (($result = $rule->parseRequest($manager, $request)) !== false) {

               if(!($rule instanceof ComposerUrlRule)){
                      Yii::trace("Request parsed with URL rule: {$rule->name}", __METHOD__);
               }

                return $result;
            }
        }

        return false;
    }
```

**Comment by samdark** [comment]
It is a community wiki article, not official documentation. Please report to article author. Thanks!

**Comment by samdark** [comment]
Do you want to use composite rules within composite rules?

**Comment by cebe** [comment]
can you give an example rules configuration?

**Comment by callmez** [comment]
@cebe same problem to me when use `yii\rest\UrlRule` within GroupUrlRule

**Comment by lynicidn** [comment]
https://github.com/yiisoft/yii2/blob/master/framework/web/GroupUrlRule.php#L48

group also composite, @cebe right u have error in config rule
`$urlManager->addRules([$groupRule])` may be u set not array ?

**Comment by lynicidn** [comment]
look it https://github.com/yiisoft/yii2/issues/3484

**Comment by dynasource** [comment]
this should be more strict indeed. Throwing an en exception for a simple trace should be prevented.

**Comment by samdark** [comment]
Won't be implemented in 2.0.
